File: functions-v4-java17/src/main/java/com/function/BlobTrigger1.java
package com.fabrikam;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.BlobTrigger;
import com.microsoft.azure.functions.annotation.BindingName;

import java.util.*;

/**
 * Azure Functions with HTTP Trigger.
 */
public class BlobTrigger1 {
    /**
     * This function listens at endpoint "/api/HttpTrigger1". Two ways to invoke it using "curl" command in bash:
     * 1. curl -d "HTTP Body" {your host}/api/HttpTrigger1
     * 2. curl {your host}/api/HttpTrigger1?name=HTTP%20Query
     */
    @FunctionName("BlobTrigger1")
    public void run(
            @BlobTrigger(name = "file",
                    dataType = "binary",
                    path = "functions-testkit-blob/{name}",
                    connection = "BlobConnectionString") byte[] content,
            @BindingName("name") String filename,
            final ExecutionContext context
    ) {
        context.getLogger().info("Name: " + filename + " Size: " + content.length + " bytes");
    }
}

File: functions-v4-typescript-modelv4/src/functions/BlobTrigger1.ts
import { app, InvocationContext } from "@azure/functions";

export async function BlobTrigger1(blob: Buffer, context: InvocationContext): Promise<void> {
    context.log(`Storage blob function processed blob "${context.triggerMetadata.name}" with size ${blob.length} bytes`);
}

app.storageBlob('BlobTrigger1', {
    path: 'samples-workitems/{name}',
    connection: '',
    handler: BlobTrigger1
});
